homeassistant/helpers/device_registry.py

```diff
@@ -12,6 +12,7 @@
 import attr
 from yarl import URL
 
+from homeassistant.backports.functools import cached_property
```

Comment:
```suggestion
from homeassistant.functools import cached_property
```
This was introduced in 3.8 does it really need backports?

Reply:
Yes because the one included in py < 3.12 has a pathological performance problem that isn't fixed until py3.12

Reply:
> The undocumented locking behavior of [cached_property()](https://docs.python.org/3.12/library/functools.html#functools.cached_property) is removed, because it locked across all instances of the class, leading to high lock contention. This means that a cached property getter function could now run more than once for a single instance, if two threads race. For most simple cached properties (e.g. those that are idempotent and simply calculate a value based on other attributes of the instance) this will be fine. If synchronization is needed, implement locking within the cached property getter function or around multi-threaded access points.

https://docs.python.org/3.12/whatsnew/3.12.html

Reply:
Aah ok